File: C#/0643. Maximum Average Subarray I.cs
public class Solution {
    public double FindMaxAverage(int[] nums, int k) {
        double rep=0;
        for(int i=0;i<k;i++){
            rep+=nums[i];
        }
        int left=0;
        int right=k;
        double result=rep;
        
        while (right<nums.Length){
            rep=rep-nums[left]+nums[right];
            result=Math.Max(result,rep);
            left+=1;
            right+=1;
        }
        
        return result/k;;
    }
}

File: 20190519python/0643. Maximum Average Subarray I.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        res=sum(nums[:k])
        rep=res/k
        
        left=0
        for i in range(k,len(nums)):
            if nums[i]>nums[left]:
                res+=nums[i]-nums[left]
                rep=max(rep,res/k)
            else:
                res+=nums[i]-nums[left]
            left+=1
        return rep
